### PR TITLE
Add OCI container building

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

**Comment by dpc** [comment]
@justinmoon I think we can land as a good start. If you can create docker hub account and put some access token into project's secrets, then I can add code that will upload release builds to it.

File: flake.nix [hunk 1]
```diff
@@ -173,6 +173,19 @@
...
+
+          container = pkgs.dockerTools.buildLayeredImage {
+            name = name;
+            contents = [ package ];
+            config = {
+              Cmd = [
+                "${package}/bin/${name}"
+              ];
+              ExposedPorts = {
+                "${builtins.toString port}/tcp" = { };
```
Comment (justinmoon) [inline flake.nix:185]: is this exposing port 8000? This `minimint` daemon currently takes 2 command line arguments: path to a config, and path to an embedded database. The config will say which port to bind to. We can change this behavior, too.

Comment (justinmoon) [reply flake.nix:185]: Also, there will be 2 ports: one for communicating with other federation servers, and another for communicating with clients

**Review by justinmoon — APPROVED** [review]

**Comment by justinmoon** [comment]
This looks good to me. Comments can be addressed in future commits. Was able to build a container on my system, too. I created a DockerHub account (can transfer to you @elsirion once we get this working). I created an "access token" and uploaded it as a github actions secret named `DOCKER_HUB_ACCESS_TOKEN`. LMK if you need anything else!

**Pull request merged**